tools/structure/src/main.py

```diff
@@ -253,7 +253,7 @@ def run(
             structured_output_dict[SettingsKeys.METADATA] = metadata
             structured_output = json.dumps(structured_output_dict)
 
-        metrics = structured_output_dict[SettingsKeys.METRICS]
+        metrics = structured_output_dict.get(SettingsKeys.METRICS, {})
```

Comment:
@Deepak-Kesavan Hope it is handled for `KeyError` in places where we use this variable when the dict is empty.

Reply:
@harini-venkataraman Yes it is handled.